packages/react/src/components/Menu/Menu.tsx

```diff
@@ -141,6 +141,17 @@ class Menu extends AutoControlledComponent<WithAsProp<MenuProps>, MenuState> {
       }
       _.invoke(predefinedProps, 'onActiveChanged', e, props)
     },
+    variables: {
+      ...variables,
+      ...predefinedProps.variables,
+    },
+  })
+
+  handleDividerOverrides = variables => predefinedProps => ({
+    variables: {
+      ...variables,
+      ...predefinedProps.variables,
```

Comment:
note that `predefinedProps.variables` may be a function (for example, if client will provide value for `variables` as a function) - in that case shallow merge logic will fail. Would suggest to reuse `mergeComponentVariables` function here - this will ensure that all `variables` formats that Stardust is/will be able to handle will be handled properly here as well, without any code changes necessary